bluemira/power_cycle/tools.py

```diff
@@ -31,3 +34,93 @@ def create_axes(ax=None):
     if ax is None:
         _, ax = plt.subplots()
     return ax
+
+
+def unique_domain(x, y, epslon=1e-10, mode="careful"):
+    """
+    Ensure x has only unique values to make (Domain: x -> Image: y) a function.
+
+    Epslon must be small enough so that consecutive values can be considered
+    equal within the context/scale in which the function is defined.
+
+    Careful mode:
+    -------------
+    Nudge forward each non-unique element in x by (N * epslon), given N times,
+    after the first appearance, that value has appeared in x before.
+
+    Fast mode:
+    ---------
+    Nudge forward every element in x by (N * epslon), with N being the index
+    of that element in x. More prone to errors if the number of elements in
+    x is large.
+    """
+    n_points = len(x)
+    if len(y) != n_points:
+        raise ValueError("x and y must have the same number of elements.")
+
+    slow_identifiers = {"careful", "c", "slow", "s"}
+    fast_identifiers = {"fast", "f"}
+
+    if mode in slow_identifiers:
+        new_x = [x[0]]
+        new_y = [y[0]]
+        nudge = 0
+        for p in range(1, n_points):
+            x_last = x[p - 1]
+            x_this = x[p]
+            y_this = y[p]
+            if np.isclose(x_last, x_this, rtol=EPS):
+                nudge += epslon
+                x_this = x_last + nudge
+            else:
+                nudge = 0
+            new_x.append(x_this)
+            new_y.append(y_this)
+
+    elif mode in fast_identifiers:
+        nudge_vector = np.arange(n_points) * epslon
+        new_x = x.copy() + nudge_vector
+        new_y = y.copy()
+
+    else:
+        raise ValueError(
+            f"Invalid argument: '{mode}'. The parameter 'mode' can only"
+            f"assume one of the following values: {slow_identifiers} or "
+            f"{fast_identifiers}."
+        )
+    return new_x, new_y
+
+
+def match_domains(
+    x_set: list[np.ndarray],
```

Comment:
got it, but domain is not a good word because the function should start with x vectors that are not necessarily a domain and make them into one... I'll think of a better name